src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -313,6 +313,266 @@ SCENARIO("Resolving entities") {
+      const openassetio::EntityReferences refs = {openassetio::EntityReference{"ref1"},
+                                                  openassetio::EntityReference{"ref2"},
+                                                  openassetio::EntityReference{"ref3"}};
+
+      const openassetio::EntityReference expected1{"expectedRef1"};
+      const openassetio::EntityReference expected2{"expectedRef2"};
+      const openassetio::EntityReference expected3{"expectedRef3"};
+      std::vector<openassetio::EntityReference> expectedVec{expected1, expected2, expected3};
+
+      // With success callback side effect, given out of order.
+      REQUIRE_CALL(*mockManagerInterfacePtr,
+                   register_(refs, threeTraitsDatas, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_5(2, expectedVec[2]))
+          .LR_SIDE_EFFECT(_5(0, expectedVec[0]))
+          .LR_SIDE_EFFECT(_5(1, expectedVec[1]));
+
+      WHEN("batch register is called with default errorPolicyTag") {
+        const std::vector<openassetio::EntityReference> actualVec =
+            manager->register_(refs, threeTraitsDatas, context);
+        THEN("returned list of EntityReferences is ordered in index order") {
+          CHECK(expectedVec == actualVec);
+        }
+      }
+      WHEN("batch register is called with kException errorPolicyTag") {
+        const std::vector<openassetio::EntityReference> actualVec =
+            manager->register_(refs, threeTraitsDatas, context,
+                               hostApi::Manager::BatchElementErrorPolicyTag::kException);
+        THEN("returned list of EntityReferences is ordered in index order") {
+          CHECK(expectedVec == actualVec);
+        }
+      }
+      WHEN("batch register is called with kVariant errorPolicyTag") {
+        std::vector<std::variant<openassetio::BatchElementError, openassetio::EntityReference>>
+            actualVec = manager->register_(refs, threeTraitsDatas, context,
+                                           hostApi::Manager::BatchElementErrorPolicyTag::kVariant);
+        THEN("returned lists of variants is ordered in index order") {
+          CHECK(expectedVec.size() == actualVec.size());
+          for (size_t i = 0; i < actualVec.size(); ++i) {
+            CHECK(std::holds_alternative<openassetio::EntityReference>(actualVec[i]));
+            auto actualVal = std::get<openassetio::EntityReference>(actualVec[i]);
+            CHECK(expectedVec[i] == actualVal);
+          }
+        }
+      }
+    }
+    GIVEN(
+        "manager plugin will encounter an entity-specific error when next registering an "
+        "EntityReference") {
+      const openassetio::EntityReference ref = openassetio::EntityReference{"testReference"};
+      const openassetio::EntityReferences refs = {ref};
+
+      openassetio::BatchElementError expected{
+          openassetio::BatchElementError::ErrorCode::kMalformedEntityReference, "Error Message"};
+
+      // With success callback side effect
```

Comment:
[fix]
```suggestion
      // With error callback side effect
```

Reply:
Fixed as part of : https://github.com/OpenAssetIO/OpenAssetIO/pull/879/commits/b17da05a778f733c17f627edce68b13b4a4a8057